AbigailDawson/knownative
# KN-2: Reusable Modal Component

### Describe your changes:

This PR creates a reusable modal component and refactors two current modals (Landing Page Warning Page and the DemoEditWordModal). The reusable modal creates a consistent style for the modal background, text, and buttons. Modal variables make adding or removing modal components (the exit "x", the delete buttons, the header title) easy.

Here is a before and after of the DemoEditWordModal.
<img width="300" src="https://github.com/user-attachments/assets/ba58be7b-741a-4daf-879b-f0ecca1ad1be">
<img width="300" src="https://github.com/user-attachments/assets/5c820a26-dde9-4231-a65c-6364854e7435">

### Issue ticket number and link:

https://huly.app/workbench/knownative/tracker/KN-2

### Notes:

The DemoEditWordModal had some icons that I did not keep in the refactored modal. If you'd like to have Icons, I think it would be good to have a conversation with either the designer or the team on icon formatting, location, and when they are necessary to keep a clean and consistent feel.

Also, the reusable modal is currently in a file called "ui-components," but I believe it should be in just "components". There is already a file called "components" that contains the Landing Page Header and Footer. I'd like to move those down into the Landing Page folder instead to avoid confusion. Let me know if that works for you.

### How to test:

Landing Page:

> Inspect/Open web console
> Click "Toggle Device Toolbar" button (or press Cmd ⌘ + Shift ⇧ + M)
> At the top, select the Dimensions dropdown, and select any option with a size smaller than 1366x1024:
> -Expected result: Modal is triggered, warning users that KnowNative is not optimized.
> User selects "Continue Anyways", triggering modal to close and user is navigated to demo
> User selects "Back", triggering modal to close and user remains on landing page

DemoEditWordModal:

> In the Demo article page, select a few words to have in your terms list. Open the Terms drawer and edit one of your saved words.
> -Expected result: Modal is triggered, displaying the "Edit Card" modal.
> User can edit the Reading and Meaning textboxes, or they can select one of the 3 buttons
> Delete button will close the modal and delete the word from the users saved words list
> Cancel will close the modal with no action taken
> Save will close the modal and save any changed made to the saved word to update the saved word list.

<sub><a href="https://huly.app/guest/knownative?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NmVhMWRjNDQ5YzY4N2Y5NWNlZTdiOWMiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWJpZ2FpbGRhd3NvLWtub3duYXRpdmUtNjYzMjgxYjEtODdiZTE2ZDY5OS0yYTZiY2QifQ.he36wCNjnL_wsI8ORq3JjaXuEUmwE-oyf1lgq3GYhCI">Huly&reg;: <b>GITHB-117</b></a></sub>
